hs/WebParsing/TimeTableParser.hs

```diff
@@ -22,10 +23,52 @@ parseTT :: IO ()
 parseTT = do
     rsp <- simpleHTTP (getRequest timetableUrl)
     body <- getResponseBody rsp
+
     let depts = getDeptList $ parseTags body
+        deptInfo = parseLinkText $ parseTags body
 
     runSqlite databasePath $
         mapM_ getDeptTimetable depts
+    runSqlite databasePath $ 
+        mapM_ insertDepartment deptInfo
+
+-- | Inserts the Department association list in the database.
+insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
+insertDepartment (code, name) = 
+    Database.Persist.Sqlite.insert_ (Tables.Department code name)
+
+-- | List of unwanted words to remove when getting department codes.
+wordsToRemove :: [String]
+wordsToRemove = ["courses",
+                "courses]", 
+                "includes",
+                "and",
+                "-",
+                "[",
+                "]",
+                ""]
+
+-- | Extracts a list of tuples containing a list of dept codes and the respective dept name.
+parseLinkText :: [Tag String] -> [([T.Text],T.Text)]
+parseLinkText tags = 
+    let tagsText = map fromTagText $ filter isTagText tags
+        containsBrackLeft = filter (\x -> '[' `elem` x) tagsText
+        convertToStr = map T.pack containsBrackLeft
+        -- deptNamesAndCodes splits lists into two strings; one containing
+        -- dept names and the other containing the unfiltered dept codes
+        deptNamesAndCodes = map (T.splitOn (T.pack "[")) convertToStr
```

Comment:
It might be more readable if the code separates defining intermediate functions from defining intermediate steps. For instance, `convertToStr` doesn't actually do anything when its declared, so it might be better to put that into a `where` clause. This way, I can understand what lines manipulate data and what lines declare functions. Just a thought.